### ISSUE TITLE
limit of atan2 expression

### ISSUE THREAD

**Issue by rtoy** [rtoy/maxima#2884]
Imported from SourceForge on 2024-07-07 00:15:23
Created by **[willisbl](https://sourceforge.net/u/willisbl/)** on 2021-09-24 12:58:48
Original: https://sourceforge.net/p/maxima/bugs/3864

---

These are OK:
~~~
(%i3)	limit(atan2(0,1-3^x),x,0, 'minus);
(%o3)	0

(%i4)	limit(atan2(0,1-3^x),x,0, 'plus);
(%o4)	%pi

~~~

But the two-sided limit isn't OK

~~~
(%i5)	limit(atan2(0,1-3^x),x,0);
atan2: atan2(0,0) is undefined.
~~~

Assuming `zeroa > 0` allows the two-sided limit to work OK:

~~~
(%i6)	assume(zeroa > 0);
(%o6)	[zeroa>0]

(%i7)	limit(atan2(0,1-3^x),x,0);
(%o7)	und
~~~

The function `simplimsubst` sometimes returns `zeroa` or `zerob`.  Later this gets converted to zero:
~~~
(%i2)	limit(atan2(0,1-3^x),x,0);
1" Enter "simplimsubst" "[zerob,atan2(0,1-3^x)
1" Exit  "simplimsubst" "false
1" Enter "simplimsubst" "[zerob,1-3^x]
1" Exit  "simplimsubst" "zeroa
atan2: atan2(0,0) is undefined. 
~~~

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-07 00:15:25
Created by **[macrakis](https://sourceforge.net/u/macrakis/)** on 2021-09-24 20:17:17
Original: https://sourceforge.net/p/maxima/bugs/3864/#9135

---

While you're in that code, you might check why it's giving **und** rather than **ind**. After all, **atan2** is bounded over its entire domain. Similarly for **limit(floor(x),x,0)**, **limit(atan(1/x),x,0)**, *etc.*

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-07 00:15:29
Created by **[dgildea](https://sourceforge.net/u/dgildea/)** on 2021-11-01 19:58:56
Original: https://sourceforge.net/p/maxima/bugs/3864/#651a

---

- **status**: open --> closed
- **assigned_to**: Dan Gildea

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-07 00:15:32
Created by **[dgildea](https://sourceforge.net/u/dgildea/)** on 2021-11-01 19:58:56
Original: https://sourceforge.net/p/maxima/bugs/3864/#7896

---

Fixed by [d9b721e55cc5207142f1ccc4b68b09a33d5f930c]
